test/functional/wallet_dust_protection.py

```
@@ -0,0 +1,257 @@
+        expected_outpoints = set()
+        for utxo in node3.listunspent():
+            if utxo['amount'] <= THRESHOLD * DUFFS:
+                expected_outpoints.add((utxo['txid'], utxo['vout']))
+        assert len(expected_outpoints) > 0, "Test requires at least one dust UTXO on node3"
+
+        # Restart node3 WITH dust protection — all existing dust should get locked
+        self.restart_node(3, ["-dustrelayfee=0", "-dustprotectionthreshold=%d" % THRESHOLD])
+        self.connect_nodes(0, 3)
+
+        locked = node3.listlockunspent()
+        locked_outpoints = {(l['txid'], l['vout']) for l in locked}
```

Flake8 (7.3.0)

[error] ambiguous variable name 'l'

(E741)